### ISSUE TITLE
Bors batch is showing the wrong status checks

### ISSUE THREAD

**Issue by RLRabinowitz** [bors-ng/bors-ng#1593]
The status check and link shown in the bors batch changes if the status check is ran again on the same commit, but for a different branch.

We have CI that is set up with github actions, in the following way:
```
name: CI

on:
  push:
    branches: ['master', 'bors']
```

So, our CI runs on both the `bors` branch and `master`. Once the bors batch is merged, CI runs again in master, on the same commit.

Once the CI runs again on `master`, bors UI gets confused.
We've had a case a status check had passed in bors, causing the PR to merge, and then the same status check failed on `master` due to a temporary glitch.
This caused bors UI to say that the batch is merged, and also show that the CI failed, causing confusion. Seems like bors is using the last run status check with the same name on the same commit status, and is not regarding the branch, and that's why it gets the `master` one
![image](https://user-images.githubusercontent.com/30493345/210509556-1ce21972-28da-4cb0-b00d-16c5144e92e2.png)

However, on the PR itself, you can see the actual status check with the correct link, to the status check run on `bors` branch
![image](https://user-images.githubusercontent.com/30493345/210509960-ee15d714-828a-45e6-a170-67c8b9fe550a.png)

So TL;DR, the status checks and links shown in bors UI's batch should point to the correct status check, also depending on the branch
